stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt

```diff
@@ -928,9 +953,14 @@ internal constructor(
         notifications.onLogout()
         // fire a handler here that the chatDomain and chatUI can use
         getCurrentUser().let(initializationCoordinator::userDisconnected)
-        socketStateService.onDisconnectRequested()
-        userStateService.onLogout()
-        socket.disconnect()
+        if (ToggleService.isSocketExperimental().not()) {
+            socketStateService.onDisconnectRequested()
+            userStateService.onLogout()
+            socket.disconnect()
+        } else {
+            userStateService.onLogout()
+            chatSocketExperimental.disconnect(DisconnectCause.UnrecoverableError(null))
```

Comment:
Same as above,  in the existing implementation, we pass DisconnectCause DisconnectCause.UnrecoverableError(null). If we change, it would be a breaking change. Do you think we should do that?

Reply:
I can't find any place on the old implementation where we are initializing a `DisconnectCause` variable with `DisconnectCause.UnrecoverableError` and a null `ChatNetworkError`.
BTW, What I mean is that it is being used on the new implementation, so it shouldn't be a breaking chage of our SDK if we use something different.
Just on this method we know it is not an `UnrecoverableError` it is a `ConnectionReleased` because the customer already asked to disconnect (`ChatClient::disconnect()`)

Reply:
So the flow previously was like -> 
1. Invoke socket.disconnect() 
2. Internal `ChatSocket` state is changed to `state = State.DisconnectedPermanently(null)`.  https://github.com/GetStream/stream-chat-android/blob/develop/stream-chat-android-client/src/main/java/io/getstream/chat/android/client/socket/ChatSocket.kt#L220
3. When state is changed to `DisconnectedPermanently`, we call `socketListener.onDisconnected(DisconnectCause.UnrecoverableError(null)` https://github.com/GetStream/stream-chat-android/blob/develop/stream-chat-android-client/src/main/java/io/getstream/chat/android/client/socket/ChatSocket.kt#L128

Reply:
Ideally, it should be `ConnectionReleased` but I doubt if any customer is doing something based on returned disconnect cause.

Reply:
Makes sense. I will change it.